joblib/test/test_parallel.py

```diff
@@ -554,3 +555,23 @@ def test_parallel_with_interactively_defined_functions():
 def test_parallel_with_exhausted_iterator():
     exhausted_iterator = iter([])
     assert_equal(Parallel(n_jobs=2)(exhausted_iterator), [])
+
+
+def check_memmap(a):
+    if not isinstance(a, np.memmap):
+        raise TypeError('Expected np.memmap instance, got %r',
+                        type(a))
+    return a
+
+
+@with_numpy
+@with_multiprocessing
+def test_auto_memmap_on_arrays_from_generator():
+    def generate_arrays(n):
```

Comment:
Maybe it would be good to explain what is going on here. Short summary + a link to this PR would be enough since this PR description is rather well written.